optimisation/design/Model_Optimisation_Design.md

```diff
@@ -253,87 +266,22 @@ length vector,
-but will enable efficient handling of resolution effects (where only the 
-![F_j](https://latex.codecogs.com/svg.latex?F_j)
-
-are convoluted with the instrumental resolution).
-
-The `OptModel` object must contain the following properties:
-
-- the multiplier, foreground, and background functions
-  + these could be stored in a single vector --
-    with a single `enum` vector or three separate vectors of indices into the functions vector indicating their types
-  + however it's probably simpler to store the different function types separately in three vectors
-- the function which evaluates the per ![\mathbb{Q}_i] parts of ![f(\mathbf{p})]
-  + some optimisers require ![f_i(\mathbf{p})] and ![\mathbf{J}_i(\mathbf{p})]
-  	so we must be able to calculate for each ![\mathbb{Q}_i] independently
-- which back-end is to be employed in the model optimisation -- an `enum` or similar, depending on the language
-- any inter-function parameter meta binding information
-- information indicating applicability of functions to datasets, when multiple datasets are fit simultaneously
-
-and must have methods which query and update its properties, plus:
-
-- simulate the composite model ![\rho(a, \ldots,\mathbf{p})](https://latex.codecogs.com/svg.latex?%5Crho%28a%2C%20%5Cldots%2C%5Cmathbf%7Bp%7D%29)
-- calculate ![r(\mathbf{p})] or the user defined ![f(\mathbf{p})]
-  - *possibly calculate vector ![f_i(\mathbf{p})] to support more back-ends* --
-  	as this requires we store the result for each ![\mathbb{Q}_i] instead of accumulating
-	the memory requirements will tend to be large.
-- estimate ![\mathbf{J}(\mathbf{p})], when required
-  - *possibly calculate the Jacobian in matrix form* which might require too much memory
-- collect the free and independent parameters from all `OptFunction`s into a single reduced parameter vector
-- distribute a reduced parameter vector to all `OptFunction`s while respecting the global parameter bindings
-- run the chosen optimisation back-end
-
-One possible implementation of the `OptModel` is
-
-![OptModel implementation diagram][OptModel_diagram]
-
-[OptModel_diagram]: images/OptModel_diagram.png 
+The use case that should be supported at minimum is that a function requires four numeric vectors as input but `OptFunction.evaluate` was provided with a `sqw` object. In this case `OptFunction.evaluate` should pull the vectors `h,k,l,E` from the `sqw` object for use by the internal function.
 
+Possible behaviour when numeric input is required but a `sqw` object is provided to `evaluate` might be
 
-## Parameter binding
-We should use functions to enforce binding of parameters.
+| required number | pass to function |
+|-----------------|------------------|
+| 0               | `void`           |
+| 1               | `E`              |
+| 2               | `norm(Q),E`      |
+| 3               | `h,k,l`          |
+| 4               | `h,k,l,E`        |
 
```

Comment:
Do we want to continue to support fitting models without direct physical interpretations?
Yes, you can fit a series of Gaussian peaks and use those to, e.g., extract a spinwave velocity **or** you could fit the peaks simultaneously as part of a model that *includes* the spinwave velocity as a parameter.

If we do want to continue to support fitting data based on its binning axes then we will need to decide on a more-complex `input_type` parameter for `OptFunction` (maybe `'coords'`, `'sqw`', and `'axes'` instead of, e.g., `'numeric'` and `'object'`)

Reply:
It was decided in the sprint meeting that we *should* continue to support fitting models without physical coordinate axes (e.g. allow fitting based on binning axes).

Reply:
Thanks, I meant to add a similar comment about the groups' decision but forgot. 😆
The newest version of the design document explicitly includes this possibility.